### PR TITLE
Fix subgroups

### PR DESCRIPTION
**Pull request by Daniel-Wang**
- Better input validation (checks that owner is alphanumeric, the empty string case comes up from a case of bad url)
- Added two subgroups tests

### CODE REVIEW

File: gitlabbot/gitlabbot/util.go [hunk 1]
```diff
@@ -84,9 +85,30 @@ func parseRepoInput(urlOrRepoPath string) (hostedURL string, repo string, err er
...
+		return false
+	}
+
+	for _, arg := range args {
+		if arg == "" {
+			return false
+		}
+	}
+
+	// check that owner is only alphanumeric
```
Comment (joshblum) [inline gitlabbot/gitlabbot/util.go]: regex?

File: gitlabbot/gitlabbot/util.go [hunk 2]
```diff
@@ -104,11 +103,6 @@ func isValidArgs (args []string) bool {
...
-	for _, r := range args[0] {
-		if !unicode.IsLower(r) && !unicode.IsNumber(r) {
-			return false
-		}
-	}
-
-	return true
-}
\ No newline at end of file
+	match, _ := regexp.MatchString("^[a-zA-Z0-9_]*$", args[0])
```
Comment (joshblum) [inline gitlabbot/gitlabbot/util.go]: can subgroups have `-` as well?

Comment (Daniel-Wang) [reply gitlabbot/gitlabbot/util.go]: yeah, here i'm only checking that the owner is alphanumeric, i can add checks for the rest of the path

File: gitlabbot/gitlabbot/util.go [hunk 3]
```diff
@@ -84,9 +84,37 @@ func parseRepoInput(urlOrRepoPath string) (hostedURL string, repo string, err er
...
+		if arg == "" {
+			return false
+		}
+
+		var regexPattern string
+		// check that owner is only alphanumeric
+		if i == 0 {
+			regexPattern = "^[a-zA-Z0-9_-]*$"
+		} else {
+			regexPattern = "^[a-zA-Z0-9_\\-\\.]*$"
```
Comment (Daniel-Wang) [inline gitlabbot/gitlabbot/util.go]: turns out i still need a slash for the dash unless it's at the end of the expression, and i needed to escape the escape '\'  lol

Comment (marceloneil) [reply gitlabbot/gitlabbot/util.go]: you can use backticks ` instead of quotes "

Comment (Daniel-Wang) [reply gitlabbot/gitlabbot/util.go]: thanks!

**Pull request merged**